client/llb/source.go

```diff
@@ -246,7 +246,10 @@ func Git(remote, ref string, opts ...GitOption) State {
 	var sshHost string
 	if protocolType == gitutil.SSHProtocol {
 		parts := strings.SplitN(remote, ":", 2)
-		if len(parts) == 2 {
+		// don't do this transformation if the text after the : looks
+		// like a port number.
+		if len(parts) == 2 &&
+			(len(parts[1]) == 0 || parts[1][0] < '0' || parts[1][0] > '9') {
```

Comment:
Would it be safer to do `strconv.ParseInt` in here?

Reply:
Why would it be safer? I don't think it's quite the same thing, for example it would parse `+22` as an int.

Reply:
iiuc this currently checks the range for first letter after `:`. Why can't a regular repo name start with a number?

Reply:
It could, but how would we know it's supposed to be a repo name and not a port number?

Reply:
I don't know how to know it precisely, but I thought it would at least be safer to check all the letters than just the first one. I didn't consider that `ParseInt` also accepts the sign.